### ISSUE TITLE
how add custom control?

### ISSUE THREAD

**Issue by hmz22** [PaulLeCam/react-leaflet#876]
Hi Dear;
Please help me for how add custom control  with useMap and useMapEvent when I have create custom control and click them map click conflict with other component map event click.
this my code:
```
import { PlusOutlined } from "@ant-design/icons";
import { message } from "antd";
import Modal from "antd/lib/modal/Modal";
import React, { useEffect, useState } from "react";
import {
  MapContainer,
  TileLayer,
  Marker,
  useMapEvents,
  useMap,
} from "react-leaflet";

const FindLocation = ({ setPosition }) => {
  const map = useMap();

  useMapEvents({
    click: (e) => {
      console.log(map);
      console.log(e);
    },
    locationfound: (e) => {
      setPosition(e.latlng);
    },
    locationerror: (e) => {
      message.error("Can't accsees location please turn on Gps first");
      console.log("ffffff");
    },
  });

  return (
    <div className="leaflet-top leaflet-right">
      <div
        className="leaflet-control leaflet-bar"
        style={{ width: 30, height: 30, background: "white" }}
      >
        <PlusOutlined
          style={{ width: "100%", height: "100%" }}
          onClick={(e) => {
            e.preventDefault();
            e.stopPropagation();
            console.log("ssss");
            map.locate();
          }}
        />
      </div>
    </div>
  );
};

const LocationMarker = ({ position, setPosition }) => {
  useMapEvents({
    click(e) {
      setPosition(e.latlng);
    },
  });

  return position === null ? null : (
    <Marker
      draggable
      position={position}
      eventHandlers={{
        dragend: (e) => {
          setPosition(e.target.getLatLng());
        },
      }}
    ></Marker>
  );
};

export default function MapComponent({
  visible,
  setVisible,
  handleOk,
  currentPosition,
}) {
  const [position, setPosition] = useState({ lat: 36.6953, lng: 53.5365 });

  useEffect(() => {
    if (currentPosition) {
      setPosition(currentPosition);
    }
  }, [currentPosition]);

  return (
    <Modal
      visible={visible}
      width={700}
      onOk={() => handleOk(position)}
      onCancel={setVisible}
    >
      <MapContainer
        center={[36.6953, 53.5365]}
        zoom={6}
        scrollWheelZoom={true}
        onClick
      >
        <TileLayer
          attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        />
        <LocationMarker position={position} setPosition={setPosition} />
        <FindLocation setPosition={setPosition} />
      </MapContainer>
    </Modal>
  );
}

```

**Comment by Lorthiz** [comment]
It's really funny that you ask for that. Issue graveyard is full of of those issues and bugs asking for the same things, my issue being closed not even a full day time ago :D

https://github.com/PaulLeCam/react-leaflet/issues/866

Here is a grat answer and I linked another issue  where was second approach I used. I already tested both and they are working amazing. If you are creating a bigger control that will also require you to block mouse wheels event add another line here: 

```
   useEffect(() => {
        if (ref.current) {
            L.DomEvent.disableClickPropagation(ref.current);
            L.DomEvent.disableScrollPropagation(ref.current);
        }
    });
```
